griptape/loaders/base_loader.py

```diff
@@ -18,8 +20,23 @@ class BaseLoader(ABC):
     def load(self, source: Any, *args, **kwargs) -> BaseArtifact | Sequence[BaseArtifact]:
         ...
 
-    @abstractmethod
     def load_collection(
-        self, sources: list[Any], *args, **kwargs
+        self, sources: Sequence[Any], *args, **kwargs
```

Comment:
We don't use sequences anywhere in the framework. Should we not introduce it here? Also, if we do introduce it, should it be `seq` instead?

Reply:
@vasinov it's already used as a return type for `BaseLoader.load`. This came as part of the pyright fixes we made recently. Curious if it's required in this particular scenario though.

Reply:
I see that list is used often, but `Sequence` is used in `ListArtifact`, `BaseLoader.load` (as @collindutter mentioned), and also (probably by mistake) in `PdfLoader.load_collection`. It seemed inconsistent, and if I had to pick one myself, I would choose `Sequence` because accepting a more general type as input gives the most freedom to the client.

(Given that `load_collection` returns a dict would never need to specifically accept a `list`, technically we don't even need an ordered collection, it just needs to be Iterable, but I think anything more general than `Sequence` is getting a little crazy.)

I'd be fine using `list` instead, but if we do go that route, then was a specific reason for `Sequence` being used in `ListArtifact`?

Reply:
The introduction of Sequence was to resolve pyright covariance issues. If we can get away with using `list` I'd say do that. But if pyright complains, `Sequence` is probably the easiest solution.

Reply:
Reverted `Sequence` back to `list`